### PR TITLE
Add support for nested objects in output

### PR DESCRIPTION
**Pull request by bdwyertech**
### Description of your changes

Add support for nested object output.

Fixes #

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

**Review by ytsarev — CHANGES_REQUESTED** [review]
Pipes are failing with `go mod tidy` please double check.

File: go.mod [hunk 1]
```diff
@@ -14,6 +14,7 @@ require (
 	golang.org/x/sync v0.1.0
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	k8s.io/api v0.26.1
+	k8s.io/apiextensions-apiserver v0.26.1
```
Comment (bobh66) [inline go.mod:17]: This might need to be v0.26.2 now that api/apimachinery/client-go have all been updated

**Review by bobh66 — CHANGES_REQUESTED** [review]
Needs a rebase and it would be good to include non-sensitive object and list outputs in the output tests to ensure that it works and that nothing breaks it in the future.

**Comment by ytsarev** [comment]
That was an accidental close. @bdwyertech Could you please rebase the PR?

**Comment by niklasbeinghaus** [comment]
any news on this? I'd like to use list outputs from the provider in a composition without hacks and I guess this is the stuff I'm waiting for...

File: go.mod (hunk 1 above)
Comment (ytsarev) [reply go.mod:18]: It seems transitive dependency set/limit 

```
go mod graph | grep apiserver
github.com/upbound/provider-terraform k8s.io/apiextensions-apiserver@v0.26.1
github.com/crossplane/crossplane-runtime@v0.20.0-rc.0.0.20230320143010-c424c4aca5b0 k8s.io/apiextensions-apiserver@v0.26.1
```

**Review by ytsarev — APPROVED** [review]
Thanks a lot for the contribution, @bdwyertech !

Tested it e2e, it looks just great 👍 

```
status:
    atProvider:
      outputs:
        subnet_data:
          arn: arn:aws:ec2:eu-west-1:12345:subnet/subnet-0481765836942e835
          id: subnet-0481765836942e835
        vpc_id: vpc-05d50d36843b95bc3
```

I've extended this PR with the associated example, linter, and pipeline dependency fixes.

@bobh66 as I added my own commits to this PR, can you please take a look before the final merge?

**Pull request merged**